google-cloud-resource_manager/test/google/cloud/resource_manager/policy_test.rb

```diff
@@ -34,4 +34,35 @@
     role.must_be :empty?
     role.frozen?.must_equal false
   end
+
+  describe :from_gapi do
+    it "creates from a typical Google::Apis::CloudresourcemanagerV1::Policy object" do
+      gapi = Google::Apis::CloudresourcemanagerV1::Policy.new(
+        etag: etag,
+        bindings: roles.map do |key, val|
+          Google::Apis::CloudresourcemanagerV1::Binding.new(
+            role: key,
+            members: val
+          )
+        end
+      )
+
+      policy = Google::Cloud::ResourceManager::Policy.from_gapi gapi
+
+      policy.must_be_kind_of Google::Cloud::ResourceManager::Policy
+      policy.etag.must_equal etag
```

Comment:
It's not really needed, but adding `policy.roles.wont_be :empty?` here would provide some "extra paranoid" security...